### PR TITLE
computer/machine frames now allow some projectiles to pass through, like girders.

### LINKED ISSUE #14623: People can use machine/computer frames to block lasers like they used to with girders.
What it says on the tin. Girders got changed to have a "girderpasschance" var that makes it possible for lasers to sometimes slip through. People now instead use machine and computer frames since they both do exactly the same thing but without that chance, albeit are a tiny bit more costly.

This can be used to fight the AI or go through the emitter maze in exactly the same way people used to.

### PR DESCRIPTION
**Pull request by Buggy123**
Fixes #14623

### CODE REVIEW

File: code/game/machinery/constructable_frame.dm [hunk 1]
```diff
@@ -12,6 +12,19 @@
 	var/list/req_components = null
 	var/list/req_component_names = null // user-friendly names of components
 	var/state = 1
+	var/machineframepasschance = 20
+
+
+/obj/structure/constructable_frame/CanPass(atom/movable/mover, turf/target, height=0)
```
Comment (ChangelingRain) [inline code/game/machinery/constructable_frame.dm]: /obj/machinery/ not /obj/structure/

File: code/game/machinery/computer/buildandrepair.dm [hunk 2]
```diff
@@ -8,8 +8,21 @@
...
+	if(height==0)
+		return 1
+	if(istype(mover) && mover.checkpass(PASSGRILLE))
+		return prob(computerframepasschance)
+	else
+		if(istype(mover, /obj/item/projectile))
+			return prob(computerframepasschance)
+		else
+			return 0
+
```
Comment (KorPhaeron) [inline code/game/machinery/computer/buildandrepair.dm]: This is all copypasted straight from girders, don't end up like me please don't copypaste.

Comment (tkdrg) [inline code/game/machinery/computer/buildandrepair.dm]: What Kor said. Make it a helper or something.

**Comment by Lumipharon** [comment]
To clarify, does this mean shots can go past computer frames, but not finished computers?

That doesn't really make sense. With machine frames and girders, there are big ass holes in them until they're finished, but with computers you just place a circuitboard and some glass in it.

I only bring it up because it would make a significant difference when it comes to shooting into/out of places such as the bridge and wardens office, which have lots of computers blocking line of fire.

**Comment by Buggy123** [comment]
Whoops I probably should have looked back here sooner. Yes, shots can go through unfinished computers, not finished computers. I could disable it I suppose but then people can still move them around and block shit.

File: code/game/machinery/computer/buildandrepair.dm (hunk 2 above)
Comment (Buggy123) [inline code/game/machinery/computer/buildandrepair.dm]: I would but I don't know what a helper is.

Comment (tkdrg) [inline code/game/machinery/computer/buildandrepair.dm]: In this context it means a global proc. basically the stuff in the __HELPERS folder (see for instance game.dm), make a new proc at the end that encapsulates this stuff and have both girders and computer frames call that.

**Comment by Lumipharon** [comment]
Just let them shoot through finished ones, I just wanted to clarify what was going on.

**Comment by Buggy123** [comment]
Oh, i gotta finish this. Right.

**Comment by KorPhaeron** [comment]
Dead PR, ask me to re-open it or make a new one if you want to work on this again

**Pull request closed without merging**